src/sentry/static/sentry/app/views/settings/components/emptyMessage.tsx

```diff
@@ -49,7 +50,25 @@ const Action = styled('div')`
   ${MarginStyles};
 `;
 
-const EmptyMessage = ({title, description, icon, children, action, size, className}) => (
+type Props = {
+  title?: React.ReactNode;
+  description?: React.ReactNode;
+  icon?: string;
+  action?: React.ReactElement;
+  size?: 'large' | 'medium';
+};
+
+type EmptyMessageProps = Omit<React.HTMLProps<HTMLDivElement>, keyof Props> & Props;
+
+const EmptyMessage = ({
+  title,
+  description,
+  icon,
+  children,
+  action,
+  size,
+  className,
+}: EmptyMessageProps) => (
```

Comment:
Actually I think what I would really like to see is for `EmptyMessage` to be the styled component itself. So this could look like this:

```
const EmptyMessage = styled(
  ({
    title,
    description,
    icon,
    children,
    action,
    size,
    ...props
  }: EmptyMessageProps) => (
    <Wrapper data-test-id="empty-message" size={size} {...props}>
      {icon && <StyledInlineSvg src={icon} size="36px" />}
      {title && <Title>{title}</Title>}
      {description && <Description>{description}</Description>}
      {children && <Description noMargin>{children}</Description>}
      {action && <Action>{action}</Action>}
    </Wrapper>
  )
)<Pick<EmptyMessageProps, 'size'>>`
  display: flex;
  text-align: center;
  align-items: center;
  flex-direction: column;
  color: ${p => p.theme.gray4};
  padding: ${p => p.theme.grid * 4}px 15%;
  font-size: ${p =>
    p.size && p.size === 'large' ? p.theme.fontSizeExtraLarge : p.theme.fontSizeLarge};
`;
```

This may not be 100% right